include/ccf/kv/untyped_map_handle.h

```diff
@@ -67,6 +67,8 @@ namespace kv::untyped
 
     // Sub-range variant of foreach. Visits the range of keys in [from, to).
     void range(
-      const ElementVisitor& fn, const KeyType& from, const KeyType& to);
+      const ElementVisitor& fn,
+      const std::optional<KeyType> from,
+      const std::optional<KeyType> to);
```

Comment:
```suggestion
      const std::optional<KeyType>& from,
      const std::optional<KeyType>& to = std::nullopt);
```

Reply:
I think that if we have default `nullopt` for either, we should have default `nullopt` for both?

Reply:
I think it could be confusing to read something like `range(map, "key1")` as the `key1` doesn't indicate whether the range starts or ends there whereas `range(map, "key1", std::nullopt)` gives a better idea of where the `key1` is used. Equally `range(map)` seems like some strangeness and something that should just compile down to `foreach` but won't and has an extra overhead.

Reply:
This is a perfect illustration of why named arguments are an excellent language feature.